src/svd.jl

```diff
@@ -245,6 +245,24 @@ svdvals(A::AbstractVector{T}) where {T} = [convert(eigtype(T), norm(A))]
 svdvals(x::Number) = abs(x)
 svdvals(S::SVD{<:Any,T}) where {T} = (S.S)::Vector{T}
 
+"""
+    rank(S::SVD{<:Any, T}; atol::Real=0, rtol::Real=min(n,m)*ϵ) where {T}
+
+Compute the numerical rank of given an SVD object by counting how many singular values are greater 
+than `max(atol, rtol*σ₁)` where `σ₁` is the largest calculated singular value.
+`atol` and `rtol` are the absolute and relative tolerances, respectively.
+The default relative tolerance is `n*ϵ`, where `n` is the size of the smallest dimension of UΣV'
+and `ϵ` is the [`eps`](@ref) of the element type of `S`.
+
+!!! compat "Julia 1.12"
+    The `rank(::SVD)` method requires at least Julia 1.12.
+"""
+function rank(S::SVD{<:Any,T}; atol::Real = 0.0, rtol::Real = (min(size(S)...)*eps(real(float(one(eltype(S))))))) where {T}
```

Comment:
It seems that `T` is not used anywhere, so this can simplified:

```suggestion
function rank(S::SVD; atol::Real = 0.0, rtol::Real = (min(size(S)...)*eps(real(float(eltype(S))))))
```

Reply:
good point, I'll remove the `T` and the call to `one` in the next commit